src/components/Sites/Sites.js

```diff
@@ -289,21 +292,38 @@ export default function Sites(props) {
     console.log('allRoutesOfUserTemp', allRoutesOfUserTemp);
 
     let idOfUserPlaces = [];
+    let counter = 0; // Initialize the counter
+    let lastItemIndexes = {}; // Initialize an object to keep track of the last index of each item
+    let updatedPlacesWithRoutes = []; // Local array to accumulate changes
 
     allRoutesOfUserTemp.forEach((route) => {
       console.log("route ---- ", route);
       route.places.forEach((item) => {
-        let temp = allPlaces.find((place) => place.id === item);// allPlaces --> AllNodePlaces
-        console.log("idOfUserPlaces", idOfUserPlaces );
-        if (temp.parent === 0 ) {//&& !idOfUserPlaces.includes(temp.id)
-          //place.parent === 0 is Site and not station
-          setAllPlacesOfUser((prevState) => prevState.concat([temp]));
-          idOfUserPlaces.push(item);
+        let temp = allPlaces.find((place) => place.id === item); // allPlaces --> AllNodePlaces
```

Comment:
// add a comment to explain what is being done here or add comment 
// TODO explain this to make it clear what we are doing here